plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.ts

```diff
@@ -284,6 +321,45 @@ export class DatabaseEntitiesCatalog implements EntitiesCatalog {
     return { toAdd, toUpdate, toIgnore };
   }
 
+  private async attachmentsHaveChanges(
```

Comment:
I don't like this, this makes checking for changes quite expensive, but as long as the processing loop isn't intelligent enough to skip emitting attachments if they don't change, we need something in here.